cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -1203,7 +1203,111 @@ subroutine stress_T   (nx_block,   ny_block,   &
       endif
 
     end subroutine stress_T
+
+    !=======================================================================
+
+! Computes divergence of stress tensor at the E or N point for the mom equation
+      
+! author: JF Lemieux, ECCC
+! Nov 2021      
+
+      subroutine div_stress  (nx_block,   ny_block,   & 
+                              ksub,       icell,     & 
+                              indxi,     indxj,     &
+                              dxE_N,   dyE_N, &
+                              dxT_U,   dyT_U, &
+                              arear,     &
+                              stressp,   stressm,     & 
+                              stress12, &
+                              F1, F2,               &
+                              grid_location)
+
+        use ice_dyn_shared, only: strain_rates_T, deformations_T, &
+                                  viscous_coeffs_and_rep_pressure_T
+        
+      integer (kind=int_kind), intent(in) :: & 
+         nx_block, ny_block, & ! block dimensions
+         ksub              , & ! subcycling step
+         icell                 ! no. of cells where epm (or npm) = 1 
+
+      integer (kind=int_kind), dimension (nx_block*ny_block), intent(in) :: &
+         indxi   , & ! compressed index in i-direction
+         indxj       ! compressed index in j-direction
+
+
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(in) :: &
+         dxE_N , & ! width of E or N-cell through the middle (m)
+         dyE_N , & ! height of E or N-cell through the middle (m)
+         dxT_U , & ! width of T or U-cell through the middle (m)
+         dyT_U , & ! height of T or U-cell through the middle (m)
+         arear    , & ! 1/earea or 1/narea
+         stressp  , & ! sigma11+sigma22
+         stressm  , & ! sigma11-sigma22
+         stress12     ! sigma12
+
+      character(len=*), intent(in) :: &
+         grid_location ! E (East) or N (North) ! TO BE IMPROVED!!!!
+      
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(out) :: &
+         F1      , & ! div of stress tensor for u component
+         F2          ! div of stress tensor for v component
+
+      ! local variables
+
+      integer (kind=int_kind) :: &
+         i, j, ij
+
+      character(len=*), parameter :: subname = '(div_stress)'
+
+!!! Instead of having the if statements below we could define for example
+!    i+ci, j+cj where ci, cj would change with grid_position
       
+      do ij = 1, icell
+         i = indxi(ij)
+         j = indxj(ij)
+
+      !-----------------------------------------------------------------
+      ! F1,F2 : div of stress tensor for u,v components
+      !-----------------------------------------------------------------
+
+         if (grid_location == 'E') then
```

Comment:
In my upcoming PR I'm using something like: 

```fortran
      select case (trim(grid_location))
         case('U')
            maxi = max(field(i,j), field(i+1,j), field(i,j+1), field(i+1,j+1))
         case('E')
            maxi = max(field(i,j), field(i+1,j))
         case('N')
            maxi = max(field(i,j), field(i,j+1))
         case default
            call abort_ice(subname // ' unkwown grid_location: ' // grid_location)
      end select
```

I think I prefer this syntax (more high-level). Also, in any case maybe we should check for `grid_location` = something else than E or N and abort ?

Reply:
I agree

Reply:
Ok I modified it.